subsys/net/ip/ipv6.c

```diff
@@ -327,6 +330,34 @@ static enum net_verdict ipv6_route_packet(struct net_pkt *pkt,
 			return NET_OK;
 		}
 	} else {
+		iface = NULL;
+
+		if (net_if_ipv6_addr_onlink(&iface, &hdr->dst)) {
+			/* The destination is reachable via iface.
+			 * But since no valid nexthop is known,
+			 * net_pkt_lladdr_dst(pkt) cannot be set here.
+			 */
+
+			net_pkt_set_orig_iface(pkt, net_pkt_iface(pkt));
+			net_pkt_set_iface(pkt, iface);
+
+			net_pkt_set_forwarding(pkt, true);
+
+			lladdr_src = net_pkt_lladdr_src(pkt);
+			lladdr_src->addr = net_pkt_lladdr_if(pkt)->addr;
+			lladdr_src->type = net_pkt_lladdr_if(pkt)->type;
+			lladdr_src->len = net_pkt_lladdr_if(pkt)->len;
+
+			ret = net_send_data(pkt);
+
+			if (ret < 0) {
+				NET_DBG("Cannot re-route pkt %p "
+					"at iface %p (%d)",
+					pkt, net_pkt_iface(pkt), ret);
+			} else {
+				return NET_OK;
+			}
+		}
 		NET_DBG("No route to %s pkt %p dropped",
```

Comment:
Here an empty line should be added before NET_DBG

Reply:
done